src/Modules/Trinity.DynamicCluster/Trinity.DynamicCluster/ChunkedStorage.cs

```diff
@@ -57,98 +188,315 @@ public override void Dispose()
 
         public override TrinityErrorCode GetCellType(long cellId, out ushort cellType)
         {
-            throw new NotImplementedException();
+            List<ushort> templist = new List<ushort>();
+
+            foreach (Storage.Storage s in PickStorages(a => Chunks[a].Covers(cellId)))
+            {
+                try
+                {
+                    TrinityErrorCode err = s.GetCellType(cellId, out cellType);
+                    templist.Add(cellType);
+                }
+                catch (Exception ex)
+                {
+                    Log.WriteLine(LogLevel.Error, "Errors occurred during GetCellType.");
+                    Log.WriteLine(LogLevel.Error, ex.ToString());
+                    cellType = 0;
+                    return TrinityErrorCode.E_FAILURE;
+                }
+            }
+            if (templist.Count == 0) { cellType = 0; return TrinityErrorCode.E_FAILURE; }
+            if (templist.Any<ushort>(_ => _ != templist[0]))
```

Comment:
Nice thinking about keeping consistent results. But I think we can further improve usability by using some error status more elaborated than E_FAILURE. For example, imagine that we have implemented `Trinity.Storage.HighAvailability.CellTypeInconsistencyException`.